File: accounts/accounts-domain/src/main/java/com/hrrm/famoney/accounts/movement/Transfer.java
package com.hrrm.famoney.accounts.movement;

import lombok.*;
import lombok.experimental.Accessors;
import org.jetbrains.annotations.NotNull;

import javax.persistence.Column;
import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;

@Entity
@DiscriminatorValue("transfer")
@Accessors(chain = true)
@Getter
@Setter
@ToString(callSuper = true)
@EqualsAndHashCode(
        onlyExplicitlyIncluded = true,
        callSuper = true
)
public class Transfer extends Movement {

    @Column(name = "opposit_account_id")
    @NotNull
    private Integer oppositAccountId;

}

File: accounts/accounts-domain/src/main/kotlin/com/hrrm/famoney/accounts/movement/Transfer.kt
package com.hrrm.famoney.accounts.movement

import javax.persistence.Column
import javax.persistence.DiscriminatorValue
import javax.persistence.Entity

@Entity
@DiscriminatorValue("transfer")
class Transfer : Movement() {
    @Column(name = "opposit_account_id")
    var oppositAccountId: Int = 0
}